RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporter.cc

```diff
@@ -84,9 +141,9 @@ void GeneralTracksImporter::importToBlock(const edm::Event& e, BlockElementImpor
     if (tk_elem != TKs_end) {
       mask[std::distance(tracks->cbegin(), track)] = false;
       // check and update if this track is a muon
-      const int muId = muAssocToTrack((*tk_elem)->trackRef(), muons);
+      const int muId = PFMuonAlgo::muAssocToTrack((*tk_elem)->trackRef(), muons);
       if (muId != -1) {
-        muonref = reco::MuonRef(muons, muId);
+        muonref = reco::MuonRef(&muons, muId);
```

Comment:
I think this muonref is not working good, but I am not sure what is the best.
`muons` used to be a handle, but now the collection in order to follow the suggested change for `PFMuonAlgo::muAssocToTrack`. Then, what is a simpler way to define `muonref`? I get always do `get` and `getHandle`, but I am not sure if that's good.

Reply:
ah, in the context where a persistency/product information is required, the handle is still more appropriate. You can not mare a Ref without a productID info which is in the Handle (or can be carried in other ways), it can not be recovered from just the pointer to the STL container.

I thought that in my suggestions in the review I was careful enough to suggest using the underlying collections only where it was clear to work; sorry if I missed some.

Reply:
I see. Thank you for the clarification.

Reply:
Probably I over-interpreted the suggestions. Now, handles are kept where I think they are still necessary.
I tested the changes with a few workflows and look ok.